### PR TITLE
fix: fall back to attachment type for extension-less filenames in AttachmentThumb

### PR DESCRIPTION
**Pull request by ephraimduncan**
## What

`AttachmentPrimitive.unstable_Thumb` (react) and `AttachmentPrimitive.Thumb` (react-native) rendered a bare `.` when the attachment filename has no extension (e.g. `README`, `Makefile`), an empty name, or a trailing dot (`report.`).

## Why

The selector split on `.` and rendered `.{ext}` unconditionally, so an empty extension produced a lone dot in the thumbnail slot. The react-ink runtime already handles this correctly by falling back to the attachment type, so this also closes a cross-runtime parity gap.

## How

- Adopted the ink logic in both primitives: a real extension (`dot > 0 && dot < name.length - 1`) renders `.ext`; otherwise the thumb renders `attachment.type` (`image` / `document` / `file`). This also treats leading-dot names (`.env`) as extension-less instead of displaying the whole name as an extension.
- Updated the react-native `AttachmentThumb` tests, which previously pinned the buggy bare-dot behavior, to cover the new contract (no extension, empty name, leading dot, trailing dot). The ink tests already pin this contract.
- Updated the behavior claims in `docs/primitives/attachment.mdx` and `docs/react-native/primitives.mdx`.
- Changeset: patch for `@assistant-ui/react` and `@assistant-ui/react-native`.

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/5421?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<a href="https://rupic.dev/s/s2.zIa2J3eKvOB8WYzR2-aZ6ScuDdQCNA7sDi_b5YZrzIYFMGw0sMXNyIybpBc?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

### CODE REVIEW

File: packages/react/src/primitives/attachment/AttachmentThumb.tsx [hunk 1]
```diff
@@ -19,13 +19,17 @@ export const AttachmentPrimitiveThumb = forwardRef<
...
-    const parts = s.attachment.name.split(".");
-    return parts.length > 1 ? parts.pop()! : "";
+  const label = useAuiState((s) => {
+    const name = s.attachment.name;
+    const dot = name.lastIndexOf(".");
+    if (dot > 0 && dot < name.length - 1) {
+      return `.${name.slice(dot + 1)}`;
+    }
+    return s.attachment.type;
   });
```
[the remark this answers is not in the document]
Comment (ephraimduncan) [reply packages/react/src/primitives/attachment/AttachmentThumb.tsx:29]: The colocated react test shipped in 9a6e0f6, so the react contract is now pinned alongside react-native and react-ink. Passing on the core hoist here: it is a cross-package design change beyond this parity fix, and divergence now fails tests in all three runtimes.

File: packages/react-native/src/primitives/attachment/AttachmentThumb.tsx [hunk 2]
```diff
@@ -5,9 +5,13 @@ import { useAuiState } from "@assistant-ui/store";
 export type AttachmentThumbProps = TextProps;
 
 export const AttachmentThumb: FC<AttachmentThumbProps> = (props) => {
-  const ext = useAuiState((s) => {
-    const parts = s.attachment.name.split(".");
-    return parts.length > 1 ? parts.pop()! : "";
+  const label = useAuiState((s) => {
```
[the remark this answers is not in the document]
Comment (ephraimduncan) [reply packages/react-native/src/primitives/attachment/AttachmentThumb.tsx:8]: Declining the hoist in this PR: cross-package design change beyond the parity fix. The contract is pinned by colocated tests in all three runtimes, so future divergence fails tests.

**Review by okisdev — APPROVED** [review]
approving. the reason this is easy is that the fix is not invented: it is the ink implementation verbatim, so this closes a parity gap rather than adding a fourth opinion about what a thumb shows.

what i ran, on top of current main rather than the branch's base: the three suites all pass, 6/6 new in react, 7/7 in react-native after the rewrite, and ink stays 16/16 untouched. i also walked the old selector through the cases before trusting the report, and it does produce a bare dot for `README`, `""` and `report.`, and it does render the whole of `.gitignore` as an extension. updating the react-native tests that pinned the bare dot instead of deleting them is the right call, and the boundary conditions hold: `dot > 0` covers the leading dot, `dot < length - 1` the trailing one, and `-1` covers both no dot and the empty name.

one thing to sequence rather than fix here. #5415 rewrites the same function in the same files, and its react-native hunk still carries the old `split(".")` selector, so once this lands that PR conflicts on exactly these lines. when you rebase it, `children ?? \`.${ext}\`` becomes `children ?? label` and the selector should not come back with it.

two notes that do not block. `attachment.type` is an open union (`"image" | "document" | "file" | (string & {})`), so an adapter with a custom type puts an arbitrary string in the thumb; that matches ink today so parity is preserved either way. and in `apps/docs/components/examples/chatgpt.tsx` the non-image branch is a `w-12` box with no truncate, where i measured `document` at 56.8px against a 48px container at `text-xs`, so that demo clone overflows for an extension-less document. `file` at 17.2px and `image` at 34.2px are fine. cosmetic, and in a docs clone rather than either published package.
